include/zephyr/drivers/kscan_matrix.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2022 Nuvoton Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief kscan_matrix public API header file.
+ *
+ * This file contains the required interface for controlling the keyboard scan
+ * matrix input/output signals.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_KB_SCAN_MATRIX_H_
+#define ZEPHYR_INCLUDE_DRIVERS_KB_SCAN_MATRIX_H_
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Drive all columns of keyboard matrix to low.
+ */
+#define KEYBOARD_COLUMN_DRIVE_ALL -2
+/**
+ * @brief Tri-state all columns of keyboard matrix.
+ */
+#define KEYBOARD_COLUMN_DRIVE_NONE -1
+
+/**
+ * @brief Keyboard scan ISR callback called when user presses to trigger the
+ * keybaord scan interrupt.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ */
+typedef void (*kscan_isr_callback_t)(const struct device *dev);
```

Comment:
One of the problems I see with this API is that we now have `kscan_isr_callback_t` and `kscan_callback_t`, which is confusing at best (more counting this API is added with no documentation at all).